web/src/CertEditPage.js

```diff
@@ -138,115 +138,152 @@ class CertEditPage extends React.Component {
-          <Col style={{marginTop: "5px"}} span={(Setting.isMobile()) ? 22 : 2}>
-            {Setting.getLabel(i18next.t("cert:Certificate"), i18next.t("cert:Certificate - Tooltip"))} :
-          </Col>
-          <Col span={editorWidth} >
-            <Button style={{marginRight: "10px", marginBottom: "10px"}} onClick={() => {
-              copy(this.state.cert.certificate);
-              Setting.showMessage("success", i18next.t("cert:Certificate copied to clipboard successfully"));
-            }}
-            >
-              {i18next.t("cert:Copy certificate")}
-            </Button>
-            <Button type="primary" onClick={() => {
-              const blob = new Blob([this.state.cert.certificate], {type: "text/plain;charset=utf-8"});
-              FileSaver.saveAs(blob, "token_jwt_key.pem");
-            }}
-            >
-              {i18next.t("cert:Download certificate")}
-            </Button>
-            <TextArea autoSize={{minRows: 30, maxRows: 30}} value={this.state.cert.certificate} onChange={e => {
-              this.updateCertField("certificate", e.target.value);
-            }} />
-          </Col>
-          <Col span={1} />
-          <Col style={{marginTop: "5px"}} span={(Setting.isMobile()) ? 22 : 2}>
-            {Setting.getLabel(i18next.t("cert:Private key"), i18next.t("cert:Private key - Tooltip"))} :
-          </Col>
-          <Col span={editorWidth} >
-            <Button style={{marginRight: "10px", marginBottom: "10px"}} onClick={() => {
-              copy(this.state.cert.privateKey);
-              Setting.showMessage("success", i18next.t("cert:Private key copied to clipboard successfully"));
-            }}
-            >
-              {i18next.t("cert:Copy private key")}
-            </Button>
-            <Button type="primary" onClick={() => {
-              const blob = new Blob([this.state.cert.privateKey], {type: "text/plain;charset=utf-8"});
-              FileSaver.saveAs(blob, "token_jwt_key.key");
-            }}
-            >
-              {i18next.t("cert:Download private key")}
-            </Button>
-            <TextArea autoSize={{minRows: 30, maxRows: 30}} value={this.state.cert.privateKey} onChange={e => {
-              this.updateCertField("privateKey", e.target.value);
-            }} />
-          </Col>
-        </Row>
+        {
+          this.state.cert.scope !== Setting.CertScopeJWT ? null : (
+            <Row style={{marginTop: "20px"}} >
+              <Col style={{marginTop: "5px"}} span={(Setting.isMobile()) ? 22 : 2}>
+                {Setting.getLabel(i18next.t("provider:Type"), i18next.t("cert:Type - Tooltip"))} :
+              </Col>
+              <Col span={22} >
+                <Select virtual={false} style={{width: "100%"}} value={this.state.cert.type} onChange={(value => {
+                  this.updateCertField("type", value);
+                })}>
+                  {
+                    [
+                      {id: "x509", name: "x509"},
```

Comment:
Isn't your cert x509?

Reply:
It is old code for JWT scope. I added additional logic for render this - jwt certificate fields (private key, cryptoAlgorithm) rendered for JWT scope.   
I don't want to renderer extra fields for CA.
If you have any ideas how to do it differently, please advise.

Reply:
If your cert is x509, you should keep this UI field and don't touch it